File: Beecrowd-Python/1047.py
# Lê a hora inicial, minuto inicial, hora final e minuto final do jogo
hora_inicial, minuto_inicial, hora_final, minuto_final = map(int, input().split())

# Calcula a duração do jogo
inicio_em_minutos = hora_inicial * 60 + minuto_inicial
final_em_minutos = hora_final * 60 + minuto_final

if inicio_em_minutos <= final_em_minutos:
    duracao_em_minutos = final_em_minutos - inicio_em_minutos
else:
    duracao_em_minutos = 24 * 60 - inicio_em_minutos + final_em_minutos

# Trata o caso especial de duração de 24 horas
if duracao_em_minutos == 0:
    duracao_horas = 24
    duracao_minutos = 0
else:
    duracao_horas = duracao_em_minutos // 60
    duracao_minutos = duracao_em_minutos % 60

# Imprime a duração do jogo no formato solicitado
print("O JOGO DUROU {} HORA(S) E {} MINUTO(S)".format(duracao_horas, duracao_minutos))

File: Beecrowd-C/1047.c
#include <stdio.h>

int main() {
    int horaInicial, minutoInicial, horaFinal, minutoFinal;
    int inicioMinutos, fimMinutos, duracao, horas, minutos;

    // Leitura da entrada
    scanf("%d %d %d %d", &horaInicial, &minutoInicial, &horaFinal, &minutoFinal);

    // Convertendo o horário inicial e final para minutos desde a meia-noite
    inicioMinutos = horaInicial * 60 + minutoInicial;
    fimMinutos = horaFinal * 60 + minutoFinal;

    // Calculando a duração do jogo em minutos
    duracao = fimMinutos - inicioMinutos;

    // Verificando se a duração é negativa (caso a hora final seja anterior à hora inicial no mesmo dia)
    if (duracao <= 0) {
        duracao += 24 * 60; // Adicionando 24 horas em minutos para considerar o jogo até o próximo dia
    }

    // Calculando horas e minutos
    horas = duracao / 60;
    minutos = duracao % 60;

    // Imprimindo o resultado
    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", horas, minutos);

    return 0;
}